doc/specs/wiki.md

```diff
@@ -0,0 +1,222 @@
+
+**Basic Flow:**
+
+1. The user navigates to the URL of the article.
+2. The article is rendered.
+
+**Functional Requirements:**
+
+* BBCode must be rendered when the article is loaded from the database.
+* The table of contents must be generated based on the pre-defined BBCode header tags.
+
+### 3.0 Editing an article
+
+**Primary Actor:** A user
+
+**Brief:** Editing articles allows mistakes and misinformation to be corrected, and current information to be expanded upon.
+
+**Preconditions:** The user has the appropriate user class to edit the article.
+
+**Trigger:** The user clicks the edit link on the article's page.
+
+**Postcondition:**
+
+* The edits to the article are saved.
+
+**Basic Flow:**
+
+1. The user clicks the edit link on the article's page.
+2. The user makes the appropriate changes to the title, body, and user class read/edit restrictions.
+3. The user clicks submit.
+4. The changes are saved to the database, and the user is redirected to the updated article.
+
+**Functional Requirements:**
+
+* The article's revision history must be updated with the changes.
+
+### 3.1 Previewing articles
+
+**Primary Actor:** A user
+
+**Brief:** When editing an article, previewing ensures that the formatting is correct.
+
+**Preconditions:** The user is creating an article, or is on the edit page of an article.
+
+**Trigger:** The user clicks the preview button.
+
+**Postcondition:**
+
+* The article is rendered in-place, without committing anything to the database.
+
+**Basic Flow:**
+
+1. The user is making changes to an article, or writing a new one.
+2. The preview button is clicked.
+3. The article is rendered in-place, and is presented to the user.
+
+**Functional Requirements:**
+
+* BBCode must be rendered when the article is previewed.
+
+### 4.0 Adding aliases
+
+**Primary Actor:** A user
+
+**Brief:** Alises allow wiki articles to be easily searched. Managing aliases ensures they are accurate and thorough.
+
+**Preconditions:** The user is currently viewing the article.
+
+**Trigger:** The user clicks the submit button next to the alias textbox.
+
+**Postcondition:**
+
+* The alias is added to the article.
+
+**Basic Flow:**
+
+1. The user enters the new alias into the textbox.
+2. The user hits submit, and the alias is added to the article.
+
+**Functional Requirements:**
+
+* Duplicate and null aliases must be able to be detected, and discarded.
+* New aliases are committed to the database.
+
+### 4.1 Removing aliases
+
+**Primary Actor:** A user
+
+**Brief:** Aliases must be able to ensure relevancy and accuracy.
+
+**Preconditions:** The user is currently viewing the article.
+
+**Trigger:** The user clicks the X button next to an alias.
+
+**Postcondition:**
+
+* The alias is removed.
+
+**Basic Flow:**
+
+1. The user clicks the X button next to an alias.
+2. The alias is removed.
+
+**Functional Requirements:**
+
+* The alias must be removed from the database.
+
+### 4.2 Viewing a user that added an alias
```

Comment:
Change this use case into "a user can view the wiki's aliases" 

Also how do aliases fit in with wiki permissions?